tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -238,124 +241,163 @@ SPECTRE_TEST_CASE("Unit.DataStructures.DataVector.MathAfterMove",
   }
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.DataVector.Math",
-                  "[Unit][DataStructures]") {
+namespace {
+enum class UseRefWrap { None, Cref, Ref };
+
+template <UseRefWrap Wrap, class T,
+          Requires<Wrap == UseRefWrap::Cref> = nullptr>
+decltype(auto) wrap(const T& t) noexcept {
+  return std::cref(t);
+}
+template <UseRefWrap Wrap, class T, Requires<Wrap == UseRefWrap::Ref> = nullptr>
+decltype(auto) wrap(T& t) noexcept {
+  return std::ref(t);
+}
+template <UseRefWrap Wrap, class T,
+          Requires<Wrap == UseRefWrap::None> = nullptr>
+decltype(auto) wrap(const T& t) noexcept {
+  return t;
+}
+
+// If Wrap is true then all values are wrapped in a std::reference_wrapper
```

Comment:
Wrap is an enum not a bool

Reply:
ping

Reply:
Sorry that I missed this :( I'm adding an explanation as to what the enum does.